answers.js

```diff
@@ -0,0 +1,107 @@
+"use strict";
+
+/*
+1. Who is bigger?
+*/
+function random(number1, number2) {
+    if (number1 < number2) {
+        return between(number1, number2)
+    } else if (number1 > number2) {
+        return between(number2, number1)
+    }
+    return number2;
+}
+
+function between(number1, number2) {
+    return number1 + (Math.random() * (number2 - number1));
+}
+
+console.log(`Random 1:10 : ${random(1,10)}`);
+console.log(`Random 10:1 : ${random(10,1)}`);
+console.log(`Random 0:100 : ${random(0,100)}`);
+console.log(`Random -500:0 : ${random(-500,0)}`);
+console.log(`Random 5:5 : ${random(5,5)}`);
+
+
+/*
+2. Sum of digits
+*/
+function sumOfDigits(number) {
+    let sum = 0
+    let absNumber = Math.abs(number);
+    while (absNumber > 0) {
+        let digit = absNumber % 10;
+        absNumber = (absNumber - digit) / 10;
+        sum+=digit;
+    }
+    return sum;
+}
+
+console.log(`Sum of digits 165 : ${sumOfDigits(165)}`);
+console.log(`Sum of digits -520 : ${sumOfDigits(520)}`);
+console.log(`Sum of digits 0 : ${sumOfDigits(0)}`);
+
+/*
+3. Guessing game
+*/
+function guessingGame() {
+    let number = Math.random() * 99 + 1;
+    const guess = prompt('Guess the number between 1 and 100 ?');
+    if (guess > number) {
+        console.log("Too big");
+    } else if (guess < number) {
+        console.log("Too small");
+    } else {
+        console.log("Bingo!");
+    }
+}
+
+console.log("You can play thte guessingGame by running guessingGame()");
+
+/*
+ROT13
+*/
+function rot13(word) {
```

Comment:
Notice this function actually does 2 things: It will both convert EACH letter to its rot13, AND does the calculations, loops, concatenation and returning the result.

I would split it into 2 functions as follows:

```
function addToCharCode(letterCode, n) {
	let base = "a".charCodeAt(0);
	let top = "z".charCodeAt(0);
	if (letterCode >= base && letterCode <= top) {
		letterCode = (letterCode + n);
		letterCode = (letterCode - base) % 26 + base;
	}
	return String.fromCharCode(letterCode);
}

function rot13(word) {
	let result = "";
	let lowerWord = word.toLowerCase();
	for (let index in lowerWord) {
		let letterCode = lowerWord.charCodeAt(index);
		result = result + addToCharCode(letterCode, 13)
	}
	return result;
}
```

As a next step I don't like this line:

```
result = result + addToCharCode(letterCode, 13)
```

Because in each loop iteration you will allocate a new string for the next result "step", and throw it away in the next iteration. When concatenating strings I much rather use Array.join as it is more efficient.

In our rot13 implementation we can add some functional style and get this:

```
function addToCharCode(n) {
  return function(char) {
    let letterCode = char.charCodeAt(0);
    let base = "a".charCodeAt(0);
    let top = "z".charCodeAt(0);
    if (letterCode >= base && letterCode <= top) {
      letterCode = (letterCode + 13);
      letterCode = (letterCode - base) % 26 + base;
    }
    return String.fromCharCode(letterCode);
  }
}

function rot13(word) {
	let result = "";
	let lowerWord = word.toLowerCase();

  return lowerWord.split('').map(addToCharCode(13)).join('');
}
```